lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -448,13 +491,48 @@ void assertConsistent() {
     }
   }
 
+  // pkg-private for testing
+  void setCountDownLatch(CountDownLatch latch) {
```

Comment:
do you think we could avoid setting a latch here, and maybe instead calling countDown from a subclass' onDocIdSetCache?